src/algorithms/simulation/simple_simulation.cpp

```diff
@@ -60,4 +60,17 @@ namespace syrec {
         }
     }
 
+    auto buildTruthTable(const qc::QuantumComputation& qc, TruthTable const& tt, std::unique_ptr<dd::Package<>>& dd) -> TruthTable {
+        TruthTable ttSimOut{};
+
+        for (auto const& [input, _]: tt) {
+            auto const inEdge    = dd->makeBasisState(static_cast<dd::Qubit>(tt.nInputs()), input.toBoolVec());
+            const auto out       = dd::simulate(&qc, inEdge, dd, 1);
+            const auto outString = out.begin()->first;
+            ttSimOut.try_emplace(input, TruthTable::Cube::fromString(outString));
+        }
+
+        return ttSimOut;
+    }
+
```

Comment:
This function should take a reversible circuit and produces a truth table. So in principle the signature would be
```c++
auto buildTruthTable(const qc::QuantumComputation& qc) -> TruthTable
```
In order to safe the copy of the return value, this might be optimized to
```c++
auto buildTruthTable(const qc::QuantumComputation& qc, TruthTable& tt) -> void
```

Please accordingly refactor this function.

Reply:
Without using tt how do I create the inEdge?

Reply:
Just how you would normally create all different `n`-bit bit strings...
Simply construct the corresponding `std::vector<bool>`s.

Reply:
tt does not contain 2^n inputs. Even if I create all the combinations, I will not be able to check since tt1 or ttOriginal does not contain those inputs.